ivadomed/uncertainty.py

```diff
@@ -70,18 +70,26 @@ def combine_predictions(fname_lst, fname_hard, fname_prob, thr=0.5):
     """
     # collect all MC simulations
     mc_data = np.array([nib.load(fname).get_fdata() for fname in fname_lst])
-    affine = nib.load(fname_lst[0]).affine
+    header = nib.load(fname_lst[0]).header
 
     # average over all the MC simulations
     data_prob = np.mean(mc_data, axis=0)
     # save prob segmentation
-    nib_prob = nib.Nifti1Image(data_prob, affine)
+    nib_prob = nib.Nifti1Image(
+        dataobj=data_prob,
+        affine=header.get_best_affine(),
+        header=header.copy()
+    )
```

Comment:
We really should be very careful and avoid keywords here (e.g. header/affine etc). Need to use much more explicit variable names and avoid keyword names for ease of reading.